prusti-specs/src/export_spec_rewriter.rs

```diff
@@ -0,0 +1,334 @@
+use proc_macro2::{TokenStream, TokenTree};
+use quote::{quote, quote_spanned, ToTokens};
+use syn::spanned::Spanned;
+
+// generate a hash of path
+fn hash_path(path: syn::Path) -> u64 {
+    use std::collections::hash_map::DefaultHasher;
+    use std::hash::{Hash, Hasher};
+    let mut hasher = DefaultHasher::new();
+    path.hash(&mut hasher);
+    hasher.finish()
+}
+
+// generate a macro identifier with given path in this format:
+// {last_segment in path}_{hash(path)}
+pub fn generate_macro_ident(path: syn::Path) -> syn::Ident {
+    let ident = path.segments.last().unwrap().ident.to_owned();
+    syn::Ident::new(&format!("{}_{}_{}", "prusti_export", ident, hash_path(path)), ident.span())
+}
+
+// generate macro for export based on given identifier and item
```

Comment:
This mostly repeats the name of the function, but doesn't really say what it does. In these macro-generating functions you might want to provide concrete examples. If it is easier, you can choose to write this in a single, large comment somewhere at the beginning of the file.